onnxruntime/python/tools/transformers/optimizer.py

```diff
@@ -174,13 +181,41 @@ def optimize_by_onnxruntime(
         else:
             providers.append("CUDAExecutionProvider")
 
-    onnxruntime.InferenceSession(onnx_model_path, sess_options, providers=providers, **kwargs)
+    if isinstance(onnx_model, str):
+        onnxruntime.InferenceSession(onnx_model, sess_options, providers=providers, **kwargs)
+    elif isinstance(onnx_model, ModelProto):
+        _load_infer_session_from_modelproto(onnx_model, save_as_external_data, sess_options, providers, kwargs)
```

Comment:
Would it be a good idea to check save_as_external_data before calling the function? Looks like the arg is only used for raising an error